### DISCUSSION TITLE
Make Granite look good on Adwaita

### DISCUSSION THREAD

**Discussion by pongloongyeat** [elementary/granite#476] [Ideas]
## Problem

Currently, some Granite widgets look quite broken with the Adwaita stylesheet (see images below). Although Granite was made to complement elementary apps, this can be extended to the Adwaita stylesheet so it will encourage more adoption of Granite and eventually (and hopefully) elementaryOS. This will introduce quite a lot of maintenance debt.

## Proposal

Make Granite widgets look good with Adwaita (or at least not broken). Note that this feature request is mostly to pull users in and achieve more adoption of use of Granite.

### Accel labels in menus

![Screenshot from 2021-03-07 14 53 13](https://user-images.githubusercontent.com/31680656/110231793-ef145680-7f54-11eb-954a-db83e35a31c7.png)

### Style Classes

![Screenshot from 2021-03-07 14-58-15](https://user-images.githubusercontent.com/31680656/110231925-8e394e00-7f55-11eb-8cb1-0d7660b9bfec.png)

### Dynamic Notebook

![Screenshot from 2021-03-07 14-58-35](https://user-images.githubusercontent.com/31680656/110231941-9db89700-7f55-11eb-8223-a53aa5e7f783.png)

### SwitchModelButton margins

![Screenshot from 2021-03-07 14-59-42](https://user-images.githubusercontent.com/31680656/110231990-c9d41800-7f55-11eb-8533-b174245fce7f.png)

### SettingsSidebar margins

![Screenshot from 2021-03-07 15-01-55](https://user-images.githubusercontent.com/31680656/110232027-1cadcf80-7f56-11eb-9ef3-bc16a08a52d2.png)

### SourceList badges

![Screenshot from 2021-03-07 15-03-15](https://user-images.githubusercontent.com/31680656/110232050-49fa7d80-7f56-11eb-9552-ed392184b474.png)

### StorageBar

![Screenshot from 2021-03-07 15-03-53](https://user-images.githubusercontent.com/31680656/110232060-6ac2d300-7f56-11eb-9bbc-ac87324649f5.png)

### Dialogs

![Screenshot from 2021-03-07 14-56-12](https://user-images.githubusercontent.com/31680656/110231857-49151c00-7f55-11eb-863c-a7d876e59cac.png)

**Comment by pongloongyeat** [comment]
As a side note, you could also publish some elementary system apps on FlatHub to showcase Granite to users/devs of other distros.

**Comment by cassidyjames** [comment]
We may be able to include _fallback_ styles in Granite widgets so they aren't totally broken on Adwaita or other stylesheets. The largest concern in my mind is indeed the maintenance burden, though, as it would either mean duplicating a lot of styles (between the Granite fallbacks and the elementary Stylesheet) or moving lots of styling out of the stylesheet and into Granite in general, plus regularly testing it on Adwaita and other popular stylesheets to ensure some amount of compatibility.

This would also mean less flexibility on our side of guaranteeing we can alter Granite styling system-wide with the stylesheet, and would require re-building and re-releasing all apps if a styling change is made.

**Reply by pongloongyeat** [reply]
> Adwaita and other popular stylesheets to ensure some amount of compatibility.

Perhaps it could just be Adwaita? Most custom themes have stylings for whatever Adwaita has and not more so I feel like Adwaita is the upper line here.

> would require re-building and re-releasing all apps

Does it really? Can you not re-release just Granite for all users and push via `apt upgrade`?

Also, if this idea does get picked up, `Granite.Dialog` could be the first to kick off and it'd be great if `Granite.Dialog` is sub-classed as a `Hdy.Window` so the headerbar wouldn't show on Adwaita. Not sure if this is the best idea but it just came to mind.

![screenshot](https://user-images.githubusercontent.com/31680656/110490850-a9f64d00-812b-11eb-8acc-b60a0e24bf15.png)

**Reply by hanaral** [reply]
I thought the idea was to use flatpak to enforce the elementary stylesheet and stop pretending that gtk=gnome
